docs/framework-java_memory_assistant.md

```diff
@@ -0,0 +1,119 @@
+This way, you should not wonder why two apparently consecutive heap dumps seem to be impossibly different from one another, e.g., singletons that are initialized in the previous heap dump have magically disappeared from the second. 
+
+###Supported Memory Areas:
+
+| Memory Area            | Property Name    |
+|------------------------|------------------|
+| Heap                   | `heap`             |
+| Code Cache             | `code_cache`       |
+| Metaspace              | `metaspace`        |
+| Compressed Class Space | `compressed_class` |
+| Eden                   | `eden`             |
+| Survivor               | `survivor`         |
+| Old Generation         | `old_gen`          |
+
+The default values can be found in the [``config/java_memory_assistant.yml``] file.
+
+###Examples:
+
+Enable the Java Memory Assistant with its default settings:
+
+```yaml
+JBP_CONFIG_JAVA_MEMORY_ASSISTANT: '{enabled : true}'
+```
+
+Create heap dumps when the heap exceeds a threshold of 75%:
+
+```yaml
+JBP_CONFIG_JAVA_MEMORY_ASSISTANT: '{enabled : true, thresholds : { heap : 75% } }'
+```
+
+Create heap dumps when the old generation grows by more than 20% in two minutes:
+
+```yaml
+JBP_CONFIG_JAVA_MEMORY_ASSISTANT: '{enabled : true, thresholds : { old_gen : +20%/2m } }'
+```
+
+### Making sure heap dumps can be created
+
+The Java Virtual Machine must create heap dumps on a file.
+Unless you are using a `filesystem service` (which does not come out of the box in a Cloud Foundry installation), it pretty much means that, even if you are uploading the heap dump somewhere else, the heap dump must first land on the ephemeral disk of the container.
+Ephemeral disks have quotas and, if all the space is taken by heap dumps (even incomplete ones!), horrible things are bound to happen to your app.
+
+The maximum size of a heap dump depends on the maximum size of the heap of the Java Virtual Machine.
+Consider increasing the disk quota of your warden/garden container via the `cf scale <app-name> -k [new size]` using as `new size` to the outcome of the following calculation:
+
+`[max heap size] * [max heap dump count] + 200MB`
+
+The aditional `200MB` is a rule-of-thumb, generous over-approximation of the amount of disk the buildpack and the application therein needs to run.
+If your application requires more filesystem than just a few tens of megabytes, you must increase the additional portion of the disk amount calculation accordingly.
+
+### Where to store heap dumps?
+
+Heap dumps are created by the Java Virtual Machine on a file on the filesystem mounted by the warden/garden container.
+Normally, the filesystem of a container is ephemeral.
+That is, if your app crashes or it is shut down, the filesystem of its container is gone with it.
+And so are your precious heap dumps.
+However, there are a few options to make sure that your heap dumps survive the decomissioning of the container.
+
+#### Container-mounted volumes
```

Comment:
I really like the idea, but I do not have an environment at my disposal to use for development and testing with the volume service. Can you or someone else give a little help there? I'll also be at Summit, in case we can have 30 mins for a quick hacking :-)